cargo/cargo_build_script_runner/bin.rs

```diff
@@ -31,18 +34,35 @@ fn main() {
     let rustc_env = env::var("RUSTC").expect("RUSTC was not set");
     // Because of the Bazel's sandbox, bazel cannot provide full path, convert all relative path to correct path.
     let manifest_dir = canonicalize(&manifest_dir_env).expect(&format!("Failed to canonicalize '{}'", manifest_dir_env));
-    let out_dir = canonicalize(&out_dir_env).expect(&format!("Failed to canonicalize '{}'", out_dir_env));
+    let out_dir =
+        canonicalize(&out_dir_env).expect(&format!("Failed to canonicalize '{}'", out_dir_env));
     let rustc = canonicalize(&rustc_env).expect(&format!("Failed to canonicalize '{}'", rustc_env));
+
+    let cc = env::var_os("CC").map(|env_var| {
+        let cc_path = Path::new(&env_var);
+        if cc_path.is_relative() {
+            exec_root.join(cc_path).into_os_string()
+        } else {
+            env_var
+        }
+    });
```

Comment:
Would replacing this with
```rust
let cc = env::var_os("CC").map(|env_var| {
  canonicalize(&env_var).expect(&format!("Failed to canonicalize '{}'", env_var))
});
```
be equivalent?  It seems like command.env takes `Path`s

Reply:
It would be equivalent assuming the `CC` variable itself isn't a symlink, and none of its parents are. I know that some remote execution platforms have options to use symlinks for input files, so you'd then end up with `CC` pointing at something like `/cas/abcd1234` rather than `/path/to/working/directory/clang` which may be problematic (e.g. if it's a script which tries to look up its own directory to find adjacent files). In generally I've trained myself to avoid `canonicalize` wherever possible - if y'all would be open to it, ideally I'd rather replace all of the `canonicalize` calls in this file with calls to `exec_root.join`, than go the other direction.

`Command::env` is happy taking anything which can be treated as an `OsStr`, and `Path` is just a wrapper around an `OsStr`, so either type is interchangeable and zero-cost here.

Reply:
Do you mind adding a comment about this?

Reply:
I just sent https://github.com/bazelbuild/rules_rust/pull/347 which has a lengthy comment - if we can get that merged I'll gladly rebase the other PRs so they pull it in :)

Reply:
Thanks for the clarification.